src/components/EventFilterColumn.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import EventFilterCategories from './EventFilterCategories';
+import './EventFilterColumn.scss';
+import _ from 'lodash/array';
+
+const EventFilterColumn = ({ events, addEventFilter, removeEventFilter }) => {
+  const eventProperties = ['location', 'type'];
+
+  function capitalize(str) {
+    return str.charAt(0).toUpperCase() + str.slice(1);
+  }
+
+  function getLocationProperty(event) {
+    return [event.values.location_address];
+  }
+
+  function getTypeProperty(event) {
+    return event.values.type.map(t => t.name);
+  }
+
+  const propertyAccessor = {
+    location: getLocationProperty,
+    type: getTypeProperty,
+  };
+
+  function allEventProperties(property) {
+    let arr = [];
+    events.forEach(e => {
+      arr.push.apply(arr, propertyAccessor[property](e));
+    });
+    return _.uniq(arr);
+  }
```

Comment:
Instead of getting the different event types through filtering the events, what if we pulled this data from the table meta data with the column info?